### PR TITLE
Sort tests in the order in which they appear in benchmarks data

### LINKED ISSUE #139: Display order of results is different than JSON
The order of the "results" (the foldable boxes) is not the same as in the emitted JSON.

There are 11 "results" in irmin's tree benchmark. The first ones are the most interesting. It would be great to have these displayed first, e.g. in the same order as in the JSON.

### PR DESCRIPTION
**Pull request by punchagan**
Closes #139

### CODE REVIEW

File: frontend/src/App.res [hunk 1]
```diff
@@ -163,8 +163,16 @@ module Benchmark = {
...
 
+    let testIndex = testName =>
+      Belt.Array.getIndexBy(data.benchmarks, b =>
+        b.test_name == testName
+      )->Belt.Option.getWithDefault(0)
+
     <Column spacing=Sx.xl>
       {graphsData
+      ->Belt.List.fromArray
+      ->Belt.List.sort(((_, _, t1), (_, _, t2)) => testIndex(t1) - testIndex(t2))
```
Comment (art-w) [inline frontend/src/App.res:174]: While it's not currently a problem with most projects having a low number of benchmarks, the `testIndex` is O(N) so the sort turns out O(N² logN). We might want to store a `test_index` property near `test_name`, or we can just make sure that all algorithms preserve the array order?

**Comment by gs0510** [comment]
@punchagan I'm not sure if this will fix the issue because currently the order in the database is completely arbitrary. So perhaps we do have to add the `test_index` because otherwise there's no way to know the order. For example here's the dune order:
```
zero_build
 dune_size
 clean_build
 zero_build
 dune_size
 dune_size
 dune_size
 clean_build
 clean_build
 zero_build
 dune_size
 dune_size
 clean_build
```

**Comment by punchagan** [comment]
Nice catches, @gs0510 and @art-w . I'll take a fresh stab at this issue, again. :)

**Comment by punchagan** [comment]
I've updated the PR to use a `test_index` to store the order in which tests are listed in the benchmarks JSON, and then sort the tests based on this value in the UI.

**Comment by gs0510** [comment]
LGTM! Thanks @punchagan!

**Pull request merged**